pkg/kubelet/lifecycle/handlers.go

```diff
@@ -132,6 +133,17 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 		}
 		resp, err := hr.httpDoer.Do(req)
 		discardHTTPRespBody(resp)
+
+		if err != nil {
+			if urlerr, ok := err.(*url.Error); ok && urlerr.Err.Error() == "http: server gave HTTP response to HTTPS client" {
+				// TODO: emit an event about the fallback
+				// TODO: increment a metric about the fallback
+				klog.V(1).ErrorS(err, "Falling back to HTTP.", "pod", klog.KObj(pod))
+				req.URL.Scheme = "http"
```

Comment:
copy the request and URL before we start modifying them, something like:

```go
httpReq := utilnet.CloneRequest(req)
httpURL := *req.URL
httpURL.Scheme = "http"
httpReq.URL = &httpURL
```

Reply:
I'd probably also drop the `Authorization` header from the req headers, just to be safe (and verify this in the HandlerFunc in the fallback unit test)

```go
httpReq.Header.Del("Authorization")
```